gnomad/variant_qc/evaluation.py

```diff
@@ -116,108 +74,78 @@ def quantiles_to_bin_boundaries(quantiles: List[int]) -> Dict:
+            hl.Struct(
                 **{
-                    bin_id: hl.tstruct(
-                        n=hl.tint64,
-                        quantiles=hl.tarray(hl.tfloat64),
-                        bin_boundaries=hl.tarray(hl.tfloat64),
-                        global_bin_indices=hl.tarray(hl.tint32),
-                        merged_bins=hl.tdict(hl.tint32, hl.tint32),
-                    )
+                    bin_id: hl.agg.filter(bin_ht[f"_filter_{bin_id}"], hl.agg.count(),)
                     for bin_id in bin_expr
                 }
-            ),
-        )
-    )
-
-    # Annotate the bin as the index in the unique boundaries array
-    bin_ht = bin_ht.annotate(
-        **{
-            bin_id: hl.or_missing(
-                bin_ht[f"_filter_{bin_id}"],
-                hl.binary_search(
-                    bin_ht.bin_stats[bin_id].bin_boundaries, bin_ht._score
-                ),
             )
-            for bin_id in bin_expr
-        }
+        )
     )
 
-    # Convert the bin to global bin by expanding merged bins, that is:
-    # If a value falls in a bin that needs expansion, assign it randomly to one of the expanded bins
-    # Otherwise, simply modify the bin to its global index (with expanded bins that is)
+    logger.info(f"Binning ranked rows into {n_bins}...")
     bin_ht = bin_ht.select(
         "snv",
         **{
-            bin_id: hl.if_else(
-                bin_ht.bin_stats[bin_id].merged_bins.contains(bin_ht[bin_id]),
-                bin_ht.bin_stats[bin_id].global_bin_indices[bin_ht[bin_id]]
-                + hl.int(
-                    hl.rand_unif(
-                        0, bin_ht.bin_stats[bin_id].merged_bins[bin_ht[bin_id]] + 1
+            bin_id: hl.int(
+                hl.floor(
+                    (
+                        n_bins
+                        * (
+                            bin_ht[f"{bin_id}_rank"]
+                            / hl.float64(bin_ht.bin_group_variant_counts[bin_id])
+                        )
                     )
-                ),
-                bin_ht.bin_stats[bin_id].global_bin_indices[bin_ht[bin_id]],
+                    + 1
+                )
             )
             for bin_id in bin_expr
         },
     )
 
     if desc:
         bin_ht = bin_ht.annotate(
-            **{bin_id: n_bins - bin_ht[bin_id] for bin_id in bin_expr}
+            **{bin_id: n_bins - bin_ht[bin_id] + 1 for bin_id in bin_expr}
         )
 
     # Because SNV and indel rows are mutually exclusive, re-combine them into a single bin.
-    # Update the global bin_stats struct to reflect the change in bin names in the table
+    # Update the global bin_group_variant_counts struct to reflect the change in bin names in the table
     if compute_snv_indel_separately:
-        bin_expr_no_snv = {bin_id.rsplit("_", 1)[0] for bin_id in bin_ht.bin_stats}
+        bin_expr_no_snv = {
+            bin_id.rsplit("_", 1)[0] for bin_id in bin_ht.rank_variant_counts
```

Comment:
I don't see this annotation defined in this script -- is this a field that's present in the input `ht`?

Reply:
oops, I changed the name after testing, but forgot one apparently